pkg/tasker/tasker.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) The Tellor Authors.
+// Licensed under the MIT License.
+
+package tasker
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/tellor-io/telliot/pkg/config"
+	"github.com/tellor-io/telliot/pkg/contracts"
+	"github.com/tellor-io/telliot/pkg/contracts/tellor"
+	"github.com/tellor-io/telliot/pkg/db"
+	"github.com/tellor-io/telliot/pkg/mining"
+	"github.com/tellor-io/telliot/pkg/rpc"
+)
+
+const ComponentName = "tasker"
+
+type Tasker struct {
+	ctx              context.Context
+	close            context.CancelFunc
+	logger           log.Logger
+	proxy            db.DataServerProxy
+	accounts         []*rpc.Account
+	contractInstance *contracts.ITellor
+	client           contracts.ETHClient
+	cfg              *config.Config
+	workSinks        map[string]chan *mining.Work
+}
+
+func CreateTasker(ctx context.Context, logger log.Logger, cfg *config.Config, proxy db.DataServerProxy, client contracts.ETHClient, contract *contracts.ITellor, accounts []*rpc.Account) (*Tasker, map[string]chan *mining.Work) {
+	ctx, close := context.WithCancel(ctx)
+	workSinks := make(map[string]chan *mining.Work)
+	for _, acc := range accounts {
+		workSinks[acc.Address.String()] = make(chan *mining.Work)
+	}
+	tasker := &Tasker{
+		ctx:              ctx,
+		close:            close,
+		proxy:            proxy,
+		accounts:         accounts,
+		contractInstance: contract,
+		workSinks:        workSinks,
+		logger:           log.With(logger, "component", ComponentName),
```

Comment:
Don't forget that for each component need to apply log filtering with `ApplyFilter`. See the other components for an example.
ApplyFilter
```suggestion
		logger:           log.With(logger, "component", ComponentName),
```